packages/@pollyjs/core/src/utils/normalize-request.js

```diff
@@ -31,18 +40,23 @@ export function headers(headers, config) {
 
   if (isObjectLike(normalizedHeaders)) {
     normalizedHeaders = new HTTPHeaders(normalizedHeaders);
+  }
 
-    // Filter out excluded headers
-    if (isObjectLike(config) && isArray(config.exclude)) {
-      config.exclude.forEach(header => (normalizedHeaders[header] = null));
-    }
+  if (isFunction(config)) {
+    normalizedHeaders = config(normalizedHeaders);
+  } else if (
+    isObjectLike(normalizedHeaders) &&
+    isObjectLike(config) &&
+    isArray(config.exclude)
+  ) {
+    config.exclude.forEach(header => delete normalizedHeaders[header]);
   }
 
   return normalizedHeaders;
 }
 
-export function body(body) {
-  return body;
+export function body(body, config) {
+  return isFunction(config) ? config(body) : body;
```

Comment:
The body passed here is the actual request body. In most cases it will be a string but it could also be a FormData, Blob, or maybe even a Buffer instance so we can't really add any preventative logic here to clone the body. 

The only solution I can think of is adding a warning in the docs about this.

Reply:
> The body passed here is the actual request body. In most cases it will be a string but it could also be a FormData, Blob, or maybe even a Buffer instance so we can't really add any preventative logic here to clone the body.

The fear here is someone mutating the body where it would effect more than just the matching comparison?

Reply:
Correct. We pass the actual request body that was originally passed in and its done before the request is sent out so if someone were to modify the Blob or FormData instance then that would be the one that would get sent out with the request. 

We can just turn this off if we think its too sketchy (along with `method`) and just support url type methods.

Reply:
> We can just turn this off if we think its too sketchy.

I see some value in being able to swap parts of the body used in the comparison (graphql requests).

Reply:
Ah right, makes perfect sense. I can just add a warning in the docs. Was there something specific you wanted to add/modify?

Reply:
Brought the conversation to slack